runtime/v2/manager.go

```diff
@@ -312,41 +312,30 @@ func (m *ShimManager) resolveRuntimePath(runtime string) (string, error) {
 		return path.(string), nil
 	}
 
-	var (
-		cmdPath string
-		lerr    error
-	)
-
-	binaryPath := shimbinary.BinaryPath(runtime)
-	if _, serr := os.Stat(binaryPath); serr == nil {
-		cmdPath = binaryPath
-	}
+	cmdPath, err := exec.LookPath(name)
+	if err != nil {
+		if eerr, ok := err.(*exec.Error); ok {
+			if eerr.Err == exec.ErrNotFound {
+				self, err := os.Executable()
+				if err != nil {
+					return "", err
+				}
 
-	if cmdPath == "" {
-		if cmdPath, lerr = exec.LookPath(name); lerr != nil {
-			if eerr, ok := lerr.(*exec.Error); ok {
-				if eerr.Err == exec.ErrNotFound {
-					self, err := os.Executable()
-					if err != nil {
-						return "", err
-					}
-
-					// Match the calling binaries (containerd) path and see
-					// if they are side by side. If so, execute the shim
-					// found there.
-					testPath := filepath.Join(filepath.Dir(self), name)
-					if _, serr := os.Stat(testPath); serr == nil {
-						cmdPath = testPath
-					}
-					if cmdPath == "" {
-						return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
-					}
+				// Match the calling binaries (containerd) path and see
+				// if they are side by side. If so, execute the shim
+				// found there.
+				testPath := filepath.Join(filepath.Dir(self), name)
+				if _, serr := os.Stat(testPath); serr == nil {
+					cmdPath = testPath
+				}
+				if cmdPath == "" {
+					return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
```

Comment:
Can this be simplified like this?
```suggestion
				if _, serr := os.Stat(testPath); serr != nil {
					return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
				}
				cmdPath = testPath
```

Reply:
If we want to optimize this part of the logic, I feel that this may be more simple and clear
```go
        cmdPath, err := exec.LookPath(name)
        if err != nil {
                if eerr, ok := err.(*exec.Error); ok && eerr.Err == exec.ErrNotFound {
                        self, err := os.Executable()
                        if err != nil {
                                return "", err
                        }

                        // Match the calling binaries (containerd) path and see
                        // if they are side by side. If so, execute the shim
                        // found there.
                        cmdPath = filepath.Join(filepath.Dir(self), name)
                        if _, err := os.Stat(cmdPath); err != nil {
                                return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
                        }
                } else {
                        return "", err
                }
        }
```
What do you think?